ModelBase/src/nodes/Reference.cpp

```diff
@@ -85,7 +85,8 @@ bool Reference::resolveHelper(bool indirect)
 
 	auto newTarget = computeTarget();
 
-	Q_ASSERT(!newTarget || (newTarget->definesSymbol() && newTarget->symbolName() == name_));
+	//TODO@cyril What is this for?
+//	Q_ASSERT(!newTarget || (newTarget->definesSymbol() && newTarget->symbolName() == name_));
```

Comment:
Hmmm, this makes sure that a reference points to a node that defines a symbol. I realize now that this doesn't necessarily hold in your case. However, for now we should not change this class.

In the Node class that you're going to create, for now, you can store a string attribute that represents the nodeID. You can get a nodeID from the treeManager: node()->manager()->nodeIdMap()->id(node()). Then later you can use this map again to map the id to the node. Node ids are of type QUuid, but that has conversions to and from string.